client/tests/simulator_tests.rs

```diff
@@ -83,6 +83,63 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random() {
```

Comment:
I have one comment here, I think this test is overkill, to my understanding we do basic statistical test for randomness. It can be simplified to making sure that the result array is not all zeros.

Also I would like to see more tests like getting zero bytes of randomness, 1 byte of randomness, and trying to get more bytes than the limits.

Reply:
Changed to test for simple repeated values test - should catch the typical memory dumps like 00 00 00 01 00 00 00 00 or typical memset patterns different from 0.